lib/park.rb

```diff
@@ -0,0 +1,25 @@
+class Park
+  attr_reader :name, :admission_price, :vehicles, :passengers, :revenue
+  def initialize(name, admission_price)
+    @name = name
+    @admission_price = admission_price
+    @vehicles = []
+    @revenue_total = 0
+  end
+
+  def add_vehicle(vehicle)
+    @vehicles << vehicle
+  end
+
+  def passengers
+    @vehicles.map do |vehicle|
+      vehicle.passengers 
+    end
+  end
+
+  def revenue
+    @vehicles.map do |vehicle|
+      @revenue_total = vehicle.passengers.count * 10
+    end
```

Comment:
The `revenue` method should return an integer, not an array. Using `map` will return an array unless we do something else to it. We can call `sum` on the end to achieve this. The other consideration is that you are hardcoding an admission fee of `10`. We should use the `admission_price` attribute so that this is dynamic and can calculate based on each park's price, which may not always be `10`. Last, the interaction pattern indicated we only charge for adults, so we don't want a count of all passengers here, just the adults.
```suggestion
    @vehicles.map do |vehicle|
      @revenue_total = vehicle.num_adults * @admission_price
    end.sum
```